### PR TITLE
fix: php/iptables support for xenial and ruby and yarn upgrades 

### PR DESCRIPTION
**Pull request by ingride**
We currently pin versions on the repo level (https://github.com/netlify/bitballoon/issues/9481) so getting Focal out of beta would mean we'd need to make some changes to improve that behavior. 
Since that is a bigger lift, what we can do to get the ball rolling is keep some of the versions in sync: ruby and yarn

This PR: 
*  removes the WARN message when installing Ruby. Ruby is actually the only message we print a warning for, which is confusing for the user since we pre-install Ruby even if the current site is not using it. https://github.com/netlify/bitballoon/issues/9481 . We do give feedback as to which version has been pinned and will be installed, and removing the WARN will align the behavior with what we do for other languages. 
* fixes an issue with the PHP PPA deprecating all PHP versions for xenial. To mitigate this, we will now install these dependencies from local packages: tracking issue: https://github.com/netlify/buildbot/issues/1419 
* moves and installs iptables from source: details: https://github.com/netlify/buildbot/issues/1419#issuecomment-861389905

This should be followed up by a bitballoon PR to align defaults for ruby and yarn https://github.com/netlify/bitballoon/pull/9908. 

Tested with PHP site: https://app.netlify.com/sites/elegant-heyrovsky-57fbb5/deploys/60c8f1d24e42f000bb017ab1

### CODE REVIEW

**Comment by rstavchansky** [comment]
This may not make sense with this particular change @ingride, but I wanted to double check whether we need updates to https://github.com/netlify/build-image/blob/xenial/included_software.md or https://github.com/netlify/build-image/blob/focal/included_software.md with this PR.

**Comment by ingride** [comment]
thanks @rstavchansky you're right! Thinking about it more now, this PR should've actually been a draft.

**Comment by verythorough** [comment]
@ingride Send me a ping if I don't notice when this gets marked ready for review! I don't expect I'll have any changes to request, aside form updating https://github.com/netlify/build-image/blob/xenial/included_software.md as Rachael noted.

**Comment by ingride** [comment]
test failures are due to PHP packages not being found. 

turns out we use `ppa:ondrej/php` to install the current PHP versions in Xenial and they pulled support for EOL Ubuntu releases https://launchpad.net/%7Eondrej/+archive/ubuntu/php/+index?batch=75&memo=150&start=150 .

**Comment by ingride** [comment]
It seems that using the current main PHP PPA (https://launchpad.net/%7Eondrej/+archive/ubuntu/php/+index?batch=75&memo=150&start=150) , we are not able to install a number of versions and packages for Xenial anymore. From their page:
```
Co-installable PHP versions: PHP 5.6, PHP 7.x and most requested extensions are included. Only Supported Versions of PHP (http://php.net/supported-versions.php) for Supported Ubuntu Releases (https://wiki.ubuntu.com/Releases) are provided. Don't ask for end-of-life PHP versions or Ubuntu release, they won't be provided.
```

The list of what is currently failing: 

```5 49.38 E: Unable to locate package php5.6-xml
#5 49.38 E: Couldn't find any package by glob 'php5.6-xml'
#5 49.38 E: Couldn't find any package by regex 'php5.6-xml'
#5 49.38 E: Unable to locate package php5.6-mbstring
#5 49.38 E: Couldn't find any package by glob 'php5.6-mbstring'
#5 49.38 E: Couldn't find any package by regex 'php5.6-mbstring'
#5 49.38 E: Unable to locate package php5.6-gd
#5 49.38 E: Couldn't find any package by glob 'php5.6-gd'
#5 49.38 E: Couldn't find any package by regex 'php5.6-gd'
#5 49.38 E: Unable to locate package php5.6-sqlite3
#5 49.38 E: Couldn't find any package by glob 'php5.6-sqlite3'
#5 49.38 E: Couldn't find any package by regex 'php5.6-sqlite3'
#5 49.38 E: Unable to locate package php5.6-curl
#5 49.38 E: Couldn't find any package by glob 'php5.6-curl'
#5 49.38 E: Couldn't find any package by regex 'php5.6-curl'
#5 49.38 E: Unable to locate package php5.6-zip
#5 49.38 E: Couldn't find any package by glob 'php5.6-zip'
#5 49.38 E: Couldn't find any package by regex 'php5.6-zip'
#5 49.38 E: Unable to locate package php5.6-intl
#5 49.38 E: Couldn't find any package by glob 'php5.6-intl'
#5 49.38 E: Couldn't find any package by regex 'php5.6-intl'
#5 49.38 E: Unable to locate package php7.2
#5 49.38 E: Couldn't find any package by glob 'php7.2'
#5 49.38 E: Couldn't find any package by regex 'php7.2'
#5 49.38 E: Unable to locate package php7.2-xml
#5 49.38 E: Couldn't find any package by glob 'php7.2-xml'
#5 49.38 E: Couldn't find any package by regex 'php7.2-xml'
#5 49.38 E: Unable to locate package php7.2-mbstring
#5 49.38 E: Couldn't find any package by glob 'php7.2-mbstring'
#5 49.38 E: Couldn't find any package by regex 'php7.2-mbstring'
#5 49.38 E: Unable to locate package php7.2-gd
#5 49.38 E: Couldn't find any package by glob 'php7.2-gd'
#5 49.38 E: Couldn't find any package by regex 'php7.2-gd'
#5 49.38 E: Unable to locate package php7.2-sqlite3
#5 49.38 E: Couldn't find any package by glob 'php7.2-sqlite3'
#5 49.38 E: Couldn't find any package by regex 'php7.2-sqlite3'
#5 49.38 E: Unable to locate package php7.2-curl
#5 49.38 E: Couldn't find any package by glob 'php7.2-curl'
#5 49.38 E: Couldn't find any package by regex 'php7.2-curl'
#5 49.38 E: Unable to locate package php7.2-zip
#5 49.38 E: Couldn't find any package by glob 'php7.2-zip'
#5 49.38 E: Couldn't find any package by regex 'php7.2-zip'
#5 49.38 E: Unable to locate package php7.2-intl
#5 49.38 E: Couldn't find any package by glob 'php7.2-intl'
#5 49.38 E: Couldn't find any package by regex 'php7.2-intl'
#5 49.38 E: Unable to locate package php7.4
#5 49.38 E: Couldn't find any package by glob 'php7.4'
#5 49.38 E: Couldn't find any package by regex 'php7.4'
#5 49.38 E: Unable to locate package php7.4-xml
#5 49.38 E: Couldn't find any package by glob 'php7.4-xml'
#5 49.38 E: Couldn't find any package by regex 'php7.4-xml'
#5 49.38 E: Unable to locate package php7.4-mbstring
#5 49.38 E: Couldn't find any package by glob 'php7.4-mbstring'
#5 49.38 E: Couldn't find any package by regex 'php7.4-mbstring'
#5 49.38 E: Unable to locate package php7.4-gd
#5 49.38 E: Couldn't find any package by glob 'php7.4-gd'
#5 49.38 E: Couldn't find any package by regex 'php7.4-gd'
#5 49.38 E: Unable to locate package php7.4-sqlite3
#5 49.38 E: Couldn't find any package by glob 'php7.4-sqlite3'
#5 49.38 E: Couldn't find any package by regex 'php7.4-sqlite3'
#5 49.38 E: Unable to locate package php7.4-curl
#5 49.38 E: Couldn't find any package by glob 'php7.4-curl'
#5 49.38 E: Couldn't find any package by regex 'php7.4-curl'
#5 49.38 E: Unable to locate package php7.4-zip
#5 49.38 E: Couldn't find any package by glob 'php7.4-zip'
#5 49.38 E: Couldn't find any package by regex 'php7.4-zip'
#5 49.38 E: Unable to locate package php7.4-intl
#5 49.38 E: Couldn't find any package by glob 'php7.4-intl'
#5 49.38 E: Couldn't find any package by regex 'php7.4-intl'
```

PHP 8.0 
```
#5 49.98 E: Unable to locate package php8.0
#5 49.98 E: Couldn't find any package by glob 'php8.0'
#5 49.98 E: Couldn't find any package by regex 'php8.0'
```

but we can install:
- php5.6
- PHP7.0 along with extension packages we used to provide before

I haven't been able to find an alternative PPA so we can install the previous versions from elsewhere, but the good news is we still offer PHP7.4 in focal, and people that need it could upgrade to that? 

cc @verythorough

File: Dockerfile [hunk 1]
```diff
@@ -140,29 +140,6 @@ RUN export DEBIAN_FRONTEND=noninteractive && \
         openjdk-8-jdk \
         optipng \
         php5.6 \
```
Comment (ingride) [inline Dockerfile]: we need this because phpbrew has a system PHP requirement https://github.com/phpbrew/phpbrew/wiki/Requirement

**Comment by ingride** [comment]
php 5.6 now installs from local sources

**Comment by ingride** [comment]
all php versions and utilities are now installed from local packages

```bash
buildbot@4d13250a1f89:/$ ls -lrta /usr/bin/ | grep php
-rwxr-xr-x 1 root root    4887776 May  1 12:52 php7.2
-rwxr-xr-x 1 root root    4492320 May  1 13:11 php5.6
-rwxr-xr-x 1 root root    4675552 May  3 11:57 php7.4
lrwxrwxrwx 1 root root         21 Jun 14 14:37 php -> /etc/alternatives/php
```

File: Dockerfile [hunk 2]
```diff
@@ -392,6 +368,22 @@ RUN boot -u
 
 USER root
 
+ADD lib/dependencies /php/dependencies/
+ADD lib/php/5.6 /php/5.6
+ADD lib/php/7.2 /php/7.2
+ADD lib/php/7.4 /php/7.4
```
Comment (mraerino) [inline Dockerfile]: does this explicitly need `ADD` or could you use `COPY` instead? i think `COPY` is generally preferrable.

File: Dockerfile [hunk 3]
```diff
@@ -392,6 +368,22 @@ RUN boot -u
...
+
+#RUN apt-get -y update && apt-get -y install libpcre3
```
Comment (mraerino) [inline Dockerfile]: leftover code?

File: Dockerfile [hunk 4]
```diff
@@ -392,6 +368,22 @@ RUN boot -u
...
+
+RUN sh -c "dpkg -i /php/dependencies/libssl1.1_1.1.1k-1+ubuntu16.04.1+deb.sury.org+0_amd64.deb"
+RUN sh -c "dpkg -i /php/dependencies/psmisc_22.21-2.1ubuntu0.1_amd64.deb"
+
+RUN export DEBIAN_FRONTEND=noninteractive && \
+    sh -c "dpkg -i /php/dependencies/*.deb" && \
+    sh -c "dpkg -i /php/5.6/*.deb" && \
+    sh -c "dpkg -i /php/7.2/*.deb" && \
+    sh -c "dpkg -i /php/7.4/*.deb"
```
Comment (mraerino) [inline Dockerfile]: i think you can omit the `sh -c` wrapping.

docs say:
> `RUN <command>`: shell form, the command is run in a shell, which by default is `/bin/sh -c` on Linux
> https://docs.docker.com/engine/reference/builder/#run

File: included_software.md [hunk 5]
```diff
@@ -20,8 +21,7 @@ The specific patch versions included will depend on when the image was last buil
   * 3.7
 * PHP - `PHP_VERSION`
   * 5.6 (default)
-  * 7.2
-  * 7.4
+  * 7.0
```
Comment (mraerino) [inline included_software.md]: what is this change about? old leftover?

File: Dockerfile (hunk 3 above)
Comment (ingride) [reply Dockerfile]: Ish, after I marked this ready for review, I spot some problems with iptables in buildbot as mentioned in the slack channel and also in [issue](https://github.com/netlify/buildbot/issues/1419), but didn't get to move the PR back to draft.

File: included_software.md (hunk 5 above)
Comment (ingride) [reply included_software.md]: yup, from when we were considering deprecating.

File: included_software.md [hunk 6]
```diff
@@ -10,6 +10,7 @@ The specific patch versions included will depend on when the image was last buil
 
 * Ruby - `RUBY_VERSION`, `.ruby-version`
   * 2.6.2 (default)
+  * 2.7.2
```
Comment (rstavchansky) [inline included_software.md:13]: @ingride Just checking whether 2.7.2 is meant to be the new default here instead of an addition.

Comment (ingride) [reply included_software.md:13]: it's meant to be a new addition, many customers depend on 2.6.2 which is why it's default.

File: Dockerfile [hunk 7]
```diff
@@ -13,9 +13,7 @@ ENV LANG en_US.UTF-8
 ENV LC_ALL en_US.UTF-8
 ENV PANDOC_VERSION 2.4
 
-# language export needed for ondrej/php PPA https://github.com/oerdnj/deb.sury.org/issues/56
-RUN export DEBIAN_FRONTEND=noninteractive && \
```
Comment (Benaiah) [inline Dockerfile:17]: Was removing this `DEBIAN_FRONTEND=noninteractive` setting an intentional change?

**Review by Benaiah — APPROVED** [review]
Assuming that removing `DEBIAN_FRONTEND=noninteractive` was intentional this looks good to me. Thanks for all your work on this!

File: Dockerfile (hunk 7 above)
Comment (ingride) [reply Dockerfile:17]: yes, this was used to make PHP logs better so we don't get `debconf: unable to initialize frontend: Readline`. I've actually moved it to the PHP installation side. 

thank you!

**Review by vbrown608 — APPROVED** [review]
Thank you!!!

**Pull request merged**